src/core/qgsattributetableconfig.h

```diff
@@ -190,6 +190,12 @@ class CORE_EXPORT QgsAttributeTableConfig
      */
     void setSortOrder( Qt::SortOrder sortOrder );
 
+    /**
+     * Compare this configuration's columns name, type, and order to \a other.
+     * Other settings (column width and visibility) are not considered.
```

Comment:
Hmm - I think visibility needs considering here. Otherwise setAttributeTableConfig will incorrectly exit early when a column is hidden before remapping.

Reply:
OK